pkg/flow/internal/importsource/import_git.go

```diff
@@ -235,16 +236,42 @@ func (im *ImportGit) pollFile(ctx context.Context, args GitArguments) error {
 		return err
 	}
 
-	// Finally, configure our controller.
-	bb, err := im.repo.ReadFile(args.Path)
+	// try to read as a directory
+	filesInfo, err := im.repo.ReadDir(args.Path)
 	if err != nil {
+		var pathErr *os.PathError
+		// if it's not a directory, then try to read as a file
+		if errors.As(err, &pathErr) && errors.Is(pathErr.Err, syscall.ENOTDIR) {
+			return im.handleFile(args.Path)
+		}
 		return err
 	}
```

Comment:
For consistency with import.file, I'd prefer if we could add a Stat method to `im.repo` and then call `im.repo.Stat(args.Path)` here to check if the path exists and if it's a file or directory.